src/roll20/content-script.js

```diff
@@ -583,7 +583,21 @@ function rollSpellAttack(request, custom_roll_dice) {
             }
         }
     }
-
+    
+    if (settings["roll20-spell-description-display"] === true) {
+        if (settings["components-display"] === "all") {
+            properties["desc"] += "\r\n\r\nDescription: " + request.description;
+        } else if (settings["components-display"] === "material") {
+            if (properties["desc"] != undefined) {
+                properties["desc"] += "\r\n\r\nDescription: " + request.description;
+            } else {
+                properties["desc"] = "Description: " + request.description;
+            }
+        } else {
+            properties["desc"] = "Description: " + request.description;
+        }
+    }
```

Comment:
I don't really understand the logic here with the `\r\n` and all that. I would say that it should instead do a display (i.e: spellcard template) before the actual roll.

Reply:
Could be done, the \r\n is to get two returns set after the Spell Components are displayed, and it makes it display in a very Roll20-reminiscent way

Reply:
Roll20 Native:
![image](https://user-images.githubusercontent.com/7988297/88459013-324f0880-ce4f-11ea-8b48-823af05317d0.png)

Beyond20 with the above code:
![image](https://user-images.githubusercontent.com/7988297/88459021-3d099d80-ce4f-11ea-88bb-c6326a054e35.png)

Reply:
Yeah, I figured `\r\n` was for adding newline (also only need `\n`), but I didn't understand the logic for why it checks for the component display and had all this logic instead of just checking if `desc` is set or not, regardless of the settings. Like : 
```js
properties["desc"] = `${properties["desc"] ? properties["desc"] + "\n\n" : ""}Description: ${request.description}`;
```

Anyway, some spells are really **huge**, and since the To-Hit is separated from the damage with that potential wall of text, I think it would be a lot less issue-prone to have it the spell card posted instead.
Something like (pseudo code in the main onMessage function) : 
```
if (roll20 description display) { 
   spell_description = displaySpell(request)
   roll = spell_description + "\n" + roll
}
```

Reply:
I think I still like my way better, because it lets you display All (Description + Components), Components only, or the regular none, and it looks super duper Roll20.